src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -889,6 +889,19 @@ private NamedTypeSymbol ConstructNamedTypeUnlessTypeArgumentOmitted(CSharpSyntax
 
             Debug.Assert(members.Count > 0);
 
+            if (!hasErrors && members[0] is LocalFunctionSymbol)
```

Comment:
It could, but the whole code path is different for local functions and normal methods. It'd just be moving this if statement somewhere else (to a place that I think makes less sense). Granted if we ever change shadowing rules, this method will be drastically refactored.